### PR TITLE
Release cleanup

### LINKED ISSUE #6431: Improvements to Versioning
There are two remaining items from #6312:

- If you bump a js package by a major version, we also bump all of its dependencies by a major version, recursively.
- If you bump the python package by major and it is not in final, do not bump the js versions, but make them match the release value.

### LINKED ISSUE #6435: Release process for alphas
The new process from #6312 is really great! I released an alpha before breakfast!

There were two things that could be adjusted:

1. The python `v1.0.0a4` tag was put on the js release commit, not the python publish commit. I think it should be on the python publish commit. I moved the tag manually for 1.0.0a4.
2. There were a few changes, like updates to yarn.lock, some package.json dependency updates, etc., that needed to be merged after the publish command was run. I amended the publish commit with those changes, and also added the hashes of the release tarballs to the commit message. See https://github.com/jupyterlab/jupyterlab/commit/05067699a16ed532977475612cecef835f5e0da3. It may be that these changes should somehow be made before the build/release/tag process.
3. I think we should separate the python build and pip upload steps to give the release manager a chance to locally install the built artifact to have a minimal sanity check before actually uploading.

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #6435, Fixes #6431

## Code changes
Cleans up the release process.
- Moves the rest of the file changes into the version process
- Adds more CI checks for versioning and integrity, including making sure expected files are present
- Moves the final tag and PyPI release to a manual process after a sanity check
- Uses the new metadata for Markdown Readmes on PyPI: https://packaging.python.org/guides/making-a-pypi-friendly-readme/

## User-facing changes
None, these are maintainer-facing.

```
$ node buildutils/lib/bump-js-major.js cells --dry-run
Would run:
lerna version premajor --preid=alpha --force-publish=cells,console,completer-extension,metapackage,console-extension,fileeditor-extension,inspector-extension,statusbar-extension,tooltip-extension,documentsearch,csvviewer-extension,documentsearch-extension,nbconvert-css,notebook,notebook-extension,vdom-extension --no-push
Done in 1.11s.
```

## Backwards-incompatible changes
None

### CODE REVIEW

**Comment by blink1073** [comment]
Note: I forgot to recurse through the deps on the major version bumper. I'd also like to add dry run flags for major version and publish.

**Comment by blink1073** [comment]
Ugh, suddenly all PRs are failing with:

```
 ERROR: Failed to download Chromium r662092! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
    Error: Download failed: server returned code 502. URL: https://storage.googleapis.com/chromium-browser-snapshots/Win_x64/662092/chrome-win.zip
        at response (D:\a\1\s\node_modules\puppeteer\lib\BrowserFetcher.js:232:21)
```

I tried setting that env variable, but it had no effect.  I'm hoping it is something temporary...

**Comment by ian-r-rose** [comment]
@blink1073 Apparently Google Cloud services are having some outages -- might it be related to that?

**Comment by blink1073** [comment]
Ah, sounds plausible. I'm not getting emails either apparently.

**Comment by blink1073** [comment]
Docs build is now the only one failing (on a Google link).  This is ready!

File: setup.py [hunk 1]
```diff
@@ -21,11 +21,9 @@
 
 NAME = 'jupyterlab'
 DESCRIPTION = 'The JupyterLab notebook server extension.'
-LONG_DESCRIPTION = """
-An extensible, comprehensive Jupyter web application.
-Development happens on https://github.com/jupyter/jupyterlab, with chat on
-https://gitter.im/jupyterlab/jupyterlab.
-"""
+
+with open(pjoin(HERE, 'README.md')) as fid:
```
Comment (ian-r-rose) [inline setup.py]: Very nice!

File: buildutils/src/bump-js-major.ts [hunk 2]
```diff
@@ -5,24 +5,77 @@
...
-        toBump.push(pkg);
+        toBump.add(pkg);
       }
+    };
+    others.forEach(pkg => {
+      maybeBump(pkg);
+    });
+
+    // Create a lut of dependencies
+    const lut: { [key: string]: { [key: string]: string } } = {};
```
Comment (jasongrout) [inline buildutils/src/bump-js-major.ts]: For whoever is wondering, lut stands for "look up table" here :)

**Pull request merged**